app/src/androidTest/java/com/github/epfl/meili/review/ReviewMenuButtonsTest.kt

```diff
@@ -0,0 +1,251 @@
+package com.github.epfl.meili.review
+
+import android.content.Intent
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import android.widget.RatingBar
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.assertion.ViewAssertions.doesNotExist
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.internal.runner.junit4.statement.UiThreadStatement
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import com.github.epfl.meili.R
+import com.github.epfl.meili.database.FirestoreDatabase
+import com.github.epfl.meili.forum.FirebasePostService
+import com.github.epfl.meili.forum.ForumViewModel
+import com.github.epfl.meili.forum.MockPostService
+import com.github.epfl.meili.forum.PostViewModel
+import com.github.epfl.meili.home.Auth
+import com.github.epfl.meili.messages.ChatMessageViewModel
+import com.github.epfl.meili.messages.MockMessageDatabase
+import com.github.epfl.meili.models.Post
+import com.github.epfl.meili.models.Post.Companion.toPost
+import com.github.epfl.meili.models.Review
+import com.google.android.gms.maps.model.LatLng
+import com.github.epfl.meili.poi.PointOfInterest
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.*
+import org.hamcrest.CoreMatchers.not
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.hamcrest.TypeSafeMatcher
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+
+
+@Suppress("UNCHECKED_CAST")
+@RunWith(AndroidJUnit4::class)
+class ReviewMenuButtonsTest {
```

Comment:
Is there a reason to include the private methods and all the mocks in this class? I don't see where they are being used.

Reply:
They're used to mock the forum activity when it's opened. I was getting internal firestore errors otherwise. By the way, you can access the chat, the review and the forum by clicking on each individual POI on the map (I'm replying here since I can't reply to the general review comment).

Reply:
I tried clicking on a POI and nothing happens.

Reply:
I removed the unused methods and mocks. If you think there are still some useless mocks please mark them using the multiple line comment feature.
As for the POI, that's weird. Does something happen when you click on the POI with the current main at least?